src/layer/AtmosphereLayer.js

```diff
@@ -0,0 +1,414 @@
+                skyData.refreshVertexBuffer = false;
+            }
+
+            gl.vertexAttribPointer(program.vertexPointLocation, 3, gl.FLOAT, false, 0, 0);
+            gl.enableVertexAttribArray(program.vertexPointLocation);
+        };
+
+        // Internal. Intentionally not documented.
+        AtmosphereLayer.prototype.applySkyIndices = function (dc) {
+
+            var gl = dc.currentGlContext,
+                skyData = this.skyData,
+                vboId;
+
+            if (!skyData.indicesVboCacheKey) {
+                skyData.indicesVboCacheKey = dc.gpuResourceCache.generateCacheKey();
+            }
+
+            vboId = dc.gpuResourceCache.resourceForKey(skyData.indicesVboCacheKey);
+            if (!vboId) {
+                vboId = gl.createBuffer();
+                dc.gpuResourceCache.putResource(skyData.indicesVboCacheKey, vboId, this._skyTriStrip.length * 2);
+                skyData.refreshIndicesBuffer = true;
+            }
+
+            gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, vboId);
+            if (skyData.refreshIndicesBuffer) {
+                gl.bufferData(gl.ELEMENT_ARRAY_BUFFER,
+                    new Uint16Array(this._skyTriStrip), gl.STATIC_DRAW);
+                dc.frameStatistics.incrementVboLoadCount(1);
+                skyData.refreshIndicesBuffer = false;
+            }
+        };
+
+        // Internal. Intentionally not documented.
+        AtmosphereLayer.prototype.drawSky = function (dc) {
+
+            var gl = dc.currentGlContext,
+                program = dc.findAndBindProgram(SkyProgram);
+
+            program.loadGlobeRadius(gl, dc.globe.equatorialRadius);
+
+            program.loadEyePoint(gl, dc.navigatorState.eyePoint);
+
+            program.loadVertexOrigin(gl, Vec3.ZERO);
+
+            program.loadModelviewProjection(gl, dc.navigatorState.modelviewProjection);
+
+            program.loadLightDirection(gl, this.activeLightDirection);
+
+            program.setScale(gl);
+
+            gl.depthMask(false);
+            gl.frontFace(gl.CW);
+
+            this.setSkyPoints(dc, program.getAltitude());
+            this.setSkyTrianglesIndices();
+
+            this.applySkyVertices(dc);
+            this.applySkyIndices(dc);
+
+            gl.drawElements(gl.TRIANGLE_STRIP, this._skyTriStrip.length, gl.UNSIGNED_SHORT, 0);
+
+            gl.depthMask(true);
+            gl.frontFace(gl.CCW);
+        };
+
+        // Internal. Intentionally not documented.
+        AtmosphereLayer.prototype.drawGround = function (dc) {
+
+            var gl = dc.currentGlContext,
+                program = dc.findAndBindProgram(GroundProgram),
+                terrain = dc.terrain,
+                textureBound;
+
+            program.loadGlobeRadius(gl, dc.globe.equatorialRadius);
+
+            program.loadEyePoint(gl, dc.navigatorState.eyePoint);
+
+            program.loadLightDirection(gl, this.activeLightDirection);
+
+            program.setScale(gl);
+
```

Comment:
It appears that calls to enableVertexAttribArray for the vertexPoint and vertexTexCoord attributes are missing here. As mentioned above, specify these vertex attribute locations explicitly at GLSL compilation time, then hard code the specified locations in our GL calls. In this case, program.vertexPointLocation can be replaced with 0 and program.vertexTexCoordLocation can be replaced with 1.